### PR TITLE
Improve swss-common rust performance by not copying string values in FFI

### PR DESCRIPTION
**Pull request by erer1243**
partner PR to https://github.com/sonic-net/sonic-swss-common/pull/935

Do we want to reorganize the files further? lmk

### CODE REVIEW

File: crates/swss-common/src/types/pod.rs [hunk 1]
```diff
@@ -0,0 +1,425 @@
+use std::{
```
Comment (r12f) [inline crates/swss-common/src/types/pod.rs:1]: do you mind to rename this file to cpp.rs?

File: crates/swss-common/src/types/class.rs [hunk 2]
```diff
@@ -0,0 +1,416 @@
+use crate::*;
+use std::{collections::HashMap, ptr::null, rc::Rc, time::Duration};
```
Comment (r12f) [inline crates/swss-common/src/types/class.rs:2]: it will be great to split this file into multiple files - one type / category for each file.

**Review by r12f — COMMENTED** [review]
overall look good to me! a few minor comments.

File: crates/swss-common/src/types/class.rs (hunk 2 above)
Comment (erer1243) [reply crates/swss-common/src/types/class.rs:2]: sure

File: crates/swss-common/src/types/pod.rs (hunk 1 above)
Comment (erer1243) [reply crates/swss-common/src/types/pod.rs:1]: I don't understand the name cpp.rs; only part of it is related to cpp

**Review by r12f — APPROVED** [review]

**Pull request merged**